tests/integration/targets/ec2/tasks/main.yml

```diff
@@ -0,0 +1,184 @@
+        name: '{{ resource_prefix }}_setup'
+        description: 'created by Ansible integration tests'
+        state: present
+        vpc_id: '{{ setup_vpc.vpc.id }}'
+      register: setup_sg
+
+    - name: Find AMI to use
+      ec2_ami_info:
+        owners: 'amazon'
+        filters:
+          name: '{{ ec2_ami_name }}'
+      register: ec2_amis
+
+    - name: Set fact with latest AMI
+      vars:
+        latest_ami: '{{ ec2_amis.images | sort(attribute="creation_date") | last }}'
+      set_fact:
+        ec2_ami_image: '{{ latest_ami.image_id }}'
+
+    # ============================================================
+
+    - name: test first instance is started
+      ec2:
+        instance_type: t2.micro
+        key_name: '{{ setup_key.key.name }}'
+        state: present
+        image: '{{ ec2_ami_image }}'
+        wait: yes
+        instance_tags:
+          ResourcePrefix: '{{ resource_prefix }}-integration_tests'
+        group_id: '{{ setup_sg.group_id }}'
+        vpc_subnet_id: '{{ setup_subnet.subnet.id }}'
+      register: test_instance_1
+
+    - name: test second instance is started
+      ec2:
+        instance_type: t2.micro
+        key_name: '{{ setup_key.key.name }}'
+        state: present
+        image: '{{ ec2_ami_image }}'
+        wait: yes
+        instance_tags:
+          ResourcePrefix: '{{ resource_prefix }}-another_tag'
+        group_id: '{{ setup_sg.group_id }}'
+        vpc_subnet_id: '{{ setup_subnet.subnet.id }}'
+      register: test_instance_2
+
+    - name: assert instances started
+      assert:
+        that:
+          - "test_instance_1.instances[0].state == 'running'"
+          - "test_instance_2.instances[0].state == 'running'"
+
+    - name: test first instance is terminated
+      ec2:
+        instance_ids: "{{ test_instance_1.instance_ids }}"
+        state: absent
+        wait: yes
+      register: result
+
+    - name: assert instance terminated
+      assert:
+        that:
+          - "result.instances[0].state == 'terminated'"
+
+    - name: test terminated instance is ignored when stopping
+      ec2:
+        instance_tags:
+          ResourcePrefix: '{{ resource_prefix }}-integration_tests'
+        state: stopped
+        wait: yes
+      register: result
+
+    - name: assert resource not changed
+      assert:
+        that:
+          - "result.changed == False"
+
+    - name: assert second instance still running
+      assert:
+        that:
+          - "test_instance_2.instances[0].state == 'running'"
+
+    # ========================================================
+
+  always:
+
+    # ============================================================
+
+
+    # TEAR DOWN: ec2 instance, ec2 key pair, security group, vpc
+    - name: Announce teardown start
+      debug:
+        msg: "***** TESTING COMPLETE. COMMENCE TEARDOWN *****"
+
+    - name: delete test instances
+      ec2:
+        instance_ids:
+          - "{{ test_instance_1.instance_ids[0] }}"
```

Comment:
Future proofing comment: If you want to reuse these values throughout the test it can be helpful to do a set_fact for them, there's a tendency in some of the different test suites for registered vars to get reused/rewritten and folks could adopt that pattern when extending these tests.  Another option is to do use ec2_instance_info in the `always` block to query all of the running instances that have the resource prefix in the tag and use that list.